strikt-arrow/strikt-arrow.gradle.kts

```diff
@@ -0,0 +1,23 @@
+import java.net.URL
+
+plugins {
+  id("java-library")
+  id("nebula.kotlin")
+  id("published")
+  id("info.solidsoft.pitest")
+}
+
+dependencies {
+  api(project(":strikt-core"))
+
+  compileOnly("io.arrow-kt:arrow-core-data:0.9.0")
```

Comment:
one open question I do have for this is how should strikt modules be aligned with arrow modules? Some data type are in other modules (like `Validate` is in `arrow-extras-data`).

An thoughts around that, or just use `compileOnly` in a similar way to what you are doing here?

Reply:
i'd tend to use _compile_ resp. _api_ scope here as i think this leaves less room for mismatches between arrow versions. 

However when doing the PR, i copied the approach from the jackson strikt matcher to be more in line with the codebase (https://github.com/robfletcher/strikt/blob/master/strikt-jackson/strikt-jackson.gradle.kts#L13)